File: elixir/rna-transcription/dna.exs
defmodule DNA do

  @thymidine ?T
  @uracil ?U

  def to_rna(dna_sequence) do
    lc nucleotide inlist dna_sequence, do: rna_counterpart(nucleotide)
  end

  defp rna_counterpart(@thymidine), do: @uracil
  defp rna_counterpart(nucleotide), do: nucleotide

end

File: javascript/rna-transcription/dna.js
module.exports = function DNA(strand) {

  var URACIL = "U";
  var THYMIDINE = "T";
  var ALL_THYMIDINES = new RegExp(THYMIDINE, "g");

  return {
    toRNA: function toRNA() {
      return strand.replace(ALL_THYMIDINES, URACIL);
    }
  };

};
